scripts/dev/configure_docker_auth.sh

```diff
@@ -45,12 +46,21 @@ check_docker_daemon_is_running
 
 if [[ -f ~/.docker/config.json ]]; then
   if [[ "${RUNNING_IN_EVG:-"false"}" != "true" ]]; then
-    # when running locally we don't need to docker login all the time - we can do it once in 11 hours (ECR tokens expire each 12 hours)
-    if [[ -n "$(find ~/.docker/config.json -mmin -360 -type f)" ]] &&
-      grep "quay.io" -q ~/.docker/config.json && # TODO to be removed at public preview stage of community-search
-      grep "268558157000" -q ~/.docker/config.json; then
-      echo "Docker credentials are up to date - not performing the new login!"
-      exit
+    # Check if login is actually required by making a HEAD request to ECR using existing Docker config
+    echo "Checking if Docker credentials are valid..."
+    ecr_auth=$(jq -r '.auths."268558157000.dkr.ecr.us-east-1.amazonaws.com".auth // empty' ~/.docker/config.json)
+
+    if [[ -n "${ecr_auth}" ]]; then
+      http_status=$(curl -s -o /dev/null -w "%{http_code}" -I --max-time 3 "https://268558157000.dkr.ecr.us-east-1.amazonaws.com/v2/dev/mongodb-kubernetes/manifests/latest" \
```

Comment:
nit: Should we use `--head` instead of `-I` to make the command more descriptive? I didn't initially understand which flag is making this request a HEAD request, but having `--head` would help.